src/Revise.jl

```diff
@@ -405,6 +413,20 @@ function parse_module!(md::ModDict, ex::Expr, file::Symbol, mod::Module, path)
     newmod
 end
 
+function watch_files_via_dir(dirname)
+    watch_file(dirname)  # this will block until there is a modification
+    latestfiles = String[]
+    wf = watched_files[dirname]
+    for file in wf.trackedfiles
+        path = joinpath(dirname, file)
+        if mtime(path) > wf.timestamp
```

Comment:
Nice. Unfortunately it seems a bit flaky, see the Travis tests (passed on 50% of the runs).